src/workerd/io/worker.c++

```diff
@@ -3948,7 +3948,27 @@ class Worker::Isolate::ResponseStreamWrapper final: public kj::AsyncOutputStream
 
   ~ResponseStreamWrapper() noexcept(false) {
     jsg::runInV8Stack([&](jsg::V8StackScope& stackScope) {
```

Comment:
If we already have a lock, then we're already in a V8 stack, s owe shouldn't call `runInV8Stack` at all I think.